### PR TITLE
[FIX] sale_order_product_recommendation: wizard lines without wizard_id reference.

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT25430
@carlosdauden @chienandalu Can you review?

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]

**Comment by sergio-teruel** [comment]
pre-commit fixed on sale_quotation_number module

**Comment by pedrobaeza** [comment]
I'm afraid there's an error in tests

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py [hunk 1]
```diff
@@ -125,6 +125,7 @@ def _generate_recommendations(self):
             i += 1
             if i >= self.line_amount:
                 break
+        recommendation_lines.wizard_id = self
```
Comment (pedrobaeza) [inline sale_order_product_recommendation/wizards/sale_order_recommendation.py]: This shouldn't be needed, as the assignation to the one2many should assign this inverse field.

Comment (chienandalu) [reply sale_order_product_recommendation/wizards/sale_order_recommendation.py]: Then maybe it's missing in the prepare: https://github.com/OCA/sale-workflow/pull/1237/files#diff-bf4bc078958d60f65a62f59551d2e6e1R65-R78

Comment (pedrobaeza) [reply sale_order_product_recommendation/wizards/sale_order_recommendation.py]: That field is filled in the next line, as part of the one2many assignation (the inverse one is assigned). If @sergio-teruel tells us the initial problem, then we can check the source of the problem.

Comment (pedrobaeza) [reply sale_order_product_recommendation/wizards/sale_order_recommendation.py]: Checked with him, and seems a fault in the ORM and he has put the more resilient way of fixing the problem and allow the method to be called from onchange or from real records. The ORM flaw might be solved with https://github.com/odoo/odoo/pull/55995 in v14.

**Comment by pedrobaeza** [comment]
/ocabot merge patch

Is there the same problem in `purchase_order_product_recommendation`?

**Pull request merged**